src/chunks/vocab/subword.rs

```diff
@@ -164,11 +165,16 @@ where
     I: Indexer,
 {
     fn subword_indices(&self, word: &str) -> Option<Vec<usize>> {
-        let indices = Self::bracket(word)
+        let word = Self::bracket(word);
+        let indices = word
             .as_str()
             .subword_indices(self.min_n as usize, self.max_n as usize, &self.indexer)
-            .map(|idx| idx as usize + self.words_len())
-            .collect::<Vec<_>>();
+            .map(|idx| idx as usize + self.words_len());
+        if I::infallible() {
```

Comment:
Did you try how well it works to always collect with capacity? Since we are able to properly give the lower and upper bounds?

Reply:
The upper bounds can be off quite a lot with the explicit indexer. We also currently don't bench explicit ngrams, so my guess is we won't observe any difference in the benches.

The lower bound is guaranteed to be 0 because of `filter_map`, so that's unfortunately of no use for us.